mplib/sapien_utils/conversion.py

```diff
@@ -81,76 +84,62 @@ def __init__(
 
         for entity in actors:
             component = entity.find_component_by_type(PhysxRigidBaseComponent)
-            assert (
-                component is not None
-            ), f"No PhysxRigidBaseComponent found in {entity.name}: {entity.components=}"
-            assert not isinstance(
-                component, PhysxArticulationLinkComponent
-            ), f"Component should not be PhysxArticulationLinkComponent: {component=}"
+            assert component is not None, (
+                f"No PhysxRigidBaseComponent found in {entity.name}: "
+                f"{entity.components=}"
+            )
 
             # Convert collision shapes at current global pose
-            col_objs = self.convert_sapien_col_shape(component)  # type: ignore
-            # TODO: multiple collision shapes
-            assert len(col_objs) == 1, (
-                f"Should only have 1 collision object, got {len(col_objs)} shapes for "
-                f"entity '{entity.name}'"
+            self.add_normal_object(
+                self.get_object_name(entity),
+                self.convert_physx_component(component),  # type: ignore
             )
 
-            self.add_normal_object(self.get_object_name(entity), col_objs[0])
-
     def update_from_simulation(self, *, update_attached_object: bool = True) -> None:
         """
-        Updates planning_world articulations/objects pose with current Scene state
+        Updates PlanningWorld's articulations/objects pose with current Scene state.
+        Note that shape's local_pose is not updated.
+        If those are changed, please recreate a new SapienPlanningWorld instance.
 
         :param update_attached_object: whether to update the attached pose of
             all attached objects
         """
         for articulation in self._sim_scene.get_all_articulations():
-            # set_qpos to update poses
-            self.get_articulation(self.get_object_name(articulation)).set_qpos(
-                articulation.qpos  # type: ignore
-            )
+            if art := self.get_articulation(self.get_object_name(articulation)):
```

Comment:
nitpicking the naming here. perhaps differentiate sapien articulation and mplib articulation

Reply:
This is fine as long as this is not external API.

Reply:
Just want to fit everything in one line as much as possible